### ISSUE TITLE
Move waypoint portion in L4 section to L7 section

### ISSUE THREAD

**Issue by linsun** [istio/istio.io#14252]
I found referring to [waypoint in L4 section](https://istio.io/latest/docs/ops/ambient/usage/ztunnel/#ztunnel-datapath-via-waypoint) is quite confusing as users may not need waypoint at all and there is no intro to it either.

I suggest moving it to waypoint section.

**Comment by linsun** [comment]
@srampal thoughts?

**Comment by husnialhamdani** [comment]
Can I work on this @linsun?

**Comment by srampal** [comment]
@***@***.*** ***@***.***> TBH I dont think this change should be
done at all. This section is not getting into any Waypoint details except
showing that sometimes ztunnel may go through Waypoint. But it is still
describing ztunnel behavior only, hence should remain in the ztunnel user's
guide.  We can chat to be clear but I dont think this change should be done
at all.

**Comment by linsun** [comment]
@srampal I'd prefer our L4 user guide to be simple, if user doesn't need waypoint, I'd rather not mention it to avoid confusing them. Right now, the user guide is quite complex for ztunnel IMHO.

**Comment by linsun** [comment]
cc @istio/wg-docs-maintainers to chime in

**Comment by frankbu** [comment]
I think I agree with @srampal that this section is only describing the ztunnel behavior so does belong in the ztunnel guide. What seems to be missing is a link to some waypoint doc (or maybe better yet, a glossary reference) for `waypoint` in this section and in the introduction. I think that would make it less confusing.

**Comment by srampal** [comment]
@frankbu Thanks. There is a separate Waypoint doc in progress by another author and once thta is published, I will add a link to it in this doc as well along with additional planned updates. 
@linsun I think this issue can be closed based on my thinking and the additional feedback from Frank .. but I leave it to you to decide.

**Comment by craigbox** [comment]
I agree with @srampal that we need to keep documentation in the ztunnel docs on how the ztunnel sends traffic to waypoints.

I also agree that these docs are far too in-depth to be in the golden path for new Ambient users.

I'm going to close this, but I'm interested in hearing where the "separate Waypoint doc in progress by another author" ended up?
